### PR TITLE
Investigation 2024 12 02 staging

### PR DESCRIPTION
**Pull request by kaladay**
This resolves the original problem described in TAMULib/fw-registry#397.

Upon upgrading to Poppy, the mapping functionality provided by external FOLIO modules does not work because the `SET_CONTRIBUTOR_TYPE_ID_BY_CODE_OR_NAME` and `TRIM_PUNCTUATION` are out of date or missing.

The `author.` is from the `100.e` from the MARC record should now be converted.
The normalization functions will strip out the punctuation, like the period `.`, and then map the name `author` into a matching UUID.

The error `22 Unprocessable Entity: "{"errors":[{"message":"must match \"^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$\"","type":"1","code":"javax.validation.constraints.Pattern.message","parameters":[{"key":"contributors[0].contributorTypeId","value":"author."}]}]}` (which does not come from mod-camunda, but instead some other module via OKAPI) no longer happens.

see: https://github.com/folio-org/data-import-processing-core/blob/42b8d49510f64694a5094a2e353af45b258ee1cf/src/main/java/org/folio/processing/mapping/defaultmapper/processor/functions/NormalizationFunction.java#L330
see: https://github.com/folio-org/data-import-processing-core/blob/42b8d49510f64694a5094a2e353af45b258ee1cf/src/main/java/org/folio/processing/mapping/defaultmapper/processor/functions/NormalizationFunction.java#L108

### CODE REVIEW

**Review by wwelling — APPROVED** [review]

File: src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+ * @see ClientHttpRequestInterceptor
+ */
+public class OkapiRestTemplate extends RestTemplate {
+
+  private static final Logger LOG = LoggerFactory.getLogger(OkapiRestTemplate.class);
+
+  /**
+   * HTTP header name for specifying the Okapi tenant identifier.
+   */
+  private static final String OKAPI_TENENT_HEADER = "X-Okapi-Tenant";
```
Comment (rmathew1011) [inline src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java:50]: OKAPI_TENANT_HEADER

**Review by rmathew1011 — CHANGES_REQUESTED** [review]
A typo - rename  OKAPI_TENENT_HEADER to OKAPI_TENANT_HEADER

File: src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java (hunk 1 above)
Comment (wwelling) [inline src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java:50]:
```suggestion
  private static final String OKAPI_TENANT_HEADER = "X-Okapi-Tenant";
```

File: src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+   * @throws IllegalArgumentException if tenant or token is null or empty
+   */
+  public OkapiRestTemplate with(@NonNull String tenant, @NonNull String token) {
+    this.setInterceptors(Collections.singletonList(new ClientHttpRequestInterceptor() {
+      @Override
+      public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttpRequestExecution execution)
+          throws IOException {
+        HttpHeaders headers = request.getHeaders();
+
+        headers.set(OKAPI_TENENT_HEADER, tenant);
```
Comment (wwelling) [inline src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java:93]:
```suggestion
        headers.set(OKAPI_TENANT_HEADER, tenant);
```

File: src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java (hunk 1 above)
Comment (wwelling) [reply src/main/java/org/folio/rest/camunda/utility/OkapiRestTemplate.java]: Done.

**Review by rmathew1011 — APPROVED** [review]

File: src/main/java/org/folio/rest/camunda/config/CamundaTenantInit.java [hunk 3]
```diff
@@ -1,26 +1,44 @@
...
 
   private static final String TENANT_TEMPLATE_KEY = "tenant";
 
+  private static String tenantHeaderName;
+
   private SqlTemplateService sqlTemplateService;
 
+  private TenantProperties tenantProperties;
+
   @Autowired
```
Comment (wwelling) [inline src/main/java/org/folio/rest/camunda/config/CamundaTenantInit.java:32]: This annotation is unnecessary.

**Review by wwelling — APPROVED** [review]

**Review by wwelling — APPROVED** [review]

**Pull request merged**